src/main/scala/org/antipathy/scoozie/testing/WorkflowTestRunner.scala

```diff
@@ -93,40 +89,38 @@ class WorkflowTestRunner(workflow: Workflow, failingNodes: Seq[String], decision
     val childNodes = Seq(fork.transitionPaths: _*)
     val simpleForkVisitor = visitor.copy(visited = visitor.visited ++ Seq(childNodes.map(n => n.action.name)))
     //failures in initial path
-    if (childNodes.map(_.action.name).exists(failingNodes.contains(_))) {
-      visitNode(childNodes.flatMap(_.failureTransition).head, simpleForkVisitor)
-    }
-    //simpleFork
-    else if (childNodes
-               .flatMap(_.successTransition)
-               .map(_.action)
-               .map {
-                 case _: Join => 1
-                 case _       => 0
-               }
-               .sum == childNodes.length) {
-      visitNode(childNodes.flatMap(_.successTransition).head, simpleForkVisitor)
-    }
-    //complex fork
-    else {
-      val outComes = for {
-        childNode <- childNodes
-        visitation = visitUntilJoin(childNode, Visitor(Seq()))
-        text = visitation.visited.flatten
-        node <- visitation.nextNodeOption
-        failed = visitation.failed
-      } yield (failed, node, text)
-
-      val outputString = Seq(Seq(s"(${outComes.map { l =>
-        l._3.mkString(" -> ")
-      }.mkString(", ")})"))
-
-      val thisVisitor = visitor.copy(visited = visitor.visited ++ outputString)
-      if (outComes.map(_._1).foldLeft(false)(_ || _)) {
-        visitNode(outComes.filter(_._1 == true).map(_._2).head, thisVisitor)
-      } else {
-        visitNode(outComes.map(_._2).head, thisVisitor)
+    val hasSimpleFailures = childNodes.map(_.action.name).exists(failingNodes.contains(_))
+    val isSimpleFork = childNodes
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Usage of untyped comparison operator](https://app.codacy.com/app/SimonJPegg/scoozie/pullRequest?prid=3072461)